libs/librepcb/libraryeditor/pkg/fsm/packageeditorstate_select.cpp

```diff
@@ -369,103 +366,125 @@ bool PackageEditorState_Select::processAbortCommand() noexcept {
  *  Private Methods
  ******************************************************************************/
 
-bool PackageEditorState_Select::openContextMenuAtPos(
-    const Point& pos) noexcept {
+bool PackageEditorState_Select::openContextMenuAtPos(const Point& pos) noexcept {
+  if (mState != SubState::IDLE) return false;
+
+  // handle item selection
+  QGraphicsItem* selectedItem = nullptr;
+  QList<QGraphicsItem*> items = findItemsAtPosition(pos);
+  if (items.isEmpty()) return false;
+  foreach (QGraphicsItem* item, items) {
+    if (item->isSelected()) {
+      selectedItem = item;
+    }
+  }
+  if (!selectedItem) {
+    clearSelectionRect(true);
+    selectedItem = items.first();
+    if (dynamic_cast<FootprintPadGraphicsItem*>(selectedItem)) {
+      // workaround for selection of a SymbolPinGraphicsItem
+      dynamic_cast<FootprintPadGraphicsItem*>(selectedItem)
+          ->setSelected(true);
+    } else {
+      selectedItem->setSelected(true);
+    }
+  }
+  Q_ASSERT(selectedItem);
+  Q_ASSERT(selectedItem->isSelected());
+
   // build the context menu
   QMenu    menu;
   QAction* aRotateCCW =
       menu.addAction(QIcon(":/img/actions/rotate_left.png"), tr("Rotate"));
+  connect(aRotateCCW, &QAction::triggered, [this](){
+    rotateSelectedItems(Angle::deg90());
+  });
   QAction* aMirrorH =
       menu.addAction(QIcon(":/img/actions/flip_horizontal.png"), tr("Mirror"));
+  connect(aMirrorH, &QAction::triggered, [this](){
+    mirrorSelectedItems(Qt::Horizontal, false);
+  });
   QAction* aFlipH = menu.addAction(QIcon(":/img/actions/swap.png"), tr("Flip"));
+  connect(aFlipH, &QAction::triggered, [this](){
+    mirrorSelectedItems(Qt::Horizontal, true);
+  });
   QAction* aRemove =
       menu.addAction(QIcon(":/img/actions/delete.png"), tr("Remove"));
+  connect(aRemove, &QAction::triggered, [this](){
+    removeSelectedItems();
+  });
   menu.addSeparator();
-  QAction* aProperties = menu.addAction(tr("Properties"));
+  QAction* aProperties =
+      menu.addAction(QIcon(":/img/actions/settings.png"), tr("Properties"));
+  connect(aProperties, &QAction::triggered, [this, &selectedItem](){
+    openPropertiesDialogOfItem(selectedItem);
+  });
 
   // execute the context menu
-  QAction* action = menu.exec(QCursor::pos());
-  if (action == aRotateCCW) {
-    return rotateSelectedItems(Angle::deg90());
-  } else if (action == aMirrorH) {
-    return mirrorSelectedItems(Qt::Horizontal, false);
-  } else if (action == aFlipH) {
-    return mirrorSelectedItems(Qt::Horizontal, true);
-  } else if (action == aRemove) {
-    return removeSelectedItems();
-  } else if (action == aProperties) {
-    return openPropertiesDialogOfItemAtPos(pos);
-  } else {
-    return false;
-  }
+  return menu.exec(QCursor::pos());
```

Comment:
If the menu was opened, the event should be considered as handled, no matter whether the user triggered one of the menu items or not. So we should always return true here.

```suggestion
  menu.exec(QCursor::pos());
  return true;
```

Reply:
As I see it, this would change the current behavior, which itself is inconsistent compared to the schematic editor.

The else statement returns false and represents the case when no action was selected.

Reply:
Hmm you're right, in the library editor it indeed would change the behavior :thinking: So I would say that was actually wrong until now :see_no_evil: 

But both the schematic editor and board editor currently always return true if the context menu was shown, no matter whether an action was clicked or not. False is only returned if the menu was not even shown. That's the correct behavior I would say.

Reply:
Ok, then I'll change it tomorrow and check for any other place where the wrong behavior might be present.